src/hypothesis/strategies.py

```diff
@@ -804,9 +804,8 @@ def dm_func(denom):
 
 @cacheable
 @defines_strategy
-def decimals(
-    min_value=None, max_value=None, allow_nan=None, allow_infinity=None
-):
+def decimals(min_value=None, max_value=None, allow_nan=None,
+             allow_infinity=None, places=None, precision=None):
```

Comment:
An analogous case where we have bounds is in the collection types, where we have min\_size and max\_size rather than allowing for a single bound. I've been thinking of adding a size parameter which has the same effect as min\_size=size, max\_size=size and is an error to mix with also providing the other two, but it's in the long tail of things I haven't quite got around to. Something like this would work here, but it would also work in the other direction, so I don't think it's actually that hard to move from one to the other.

Another point in favour of bounds is that we can potentially arrange things to shrink better than we'd get if the user did it themselves with flatmap - doing it with flatmap will probably shrink OK but not great. Even if we just use flatmap internally for now it opens up the option to change that later if and when it becomes a problem in practice. 

So my inclination is marginally in favour of bounds with a possible optional fixed size parameter as a convenience feature, but I haven't thought about the problem domain in any great depth and this should definitely be taken as inclination and not an official decree.